tests/unit/data_processing/test_data_processor.py

```diff
@@ -1341,6 +1341,79 @@ def test__create_config(self):
+            - If the column is of sdtype ``unknown`` or ``id``, then they have their specific
+            logic, independent of the ``pii`` key.
+            - Otherwise, the ``pii`` key is set to ``True`` by default. If it's True, the column is
+            assigned to AnonymizedFaker. If it's False, the column is assigned to the default
+            categorical transformer.
+        """
+        data = pd.DataFrame({
+            'name_pii': ['John', 'Doe', 'Johanna'],
+            'phone_pii': ['123-456-7890', '123-456-7890', '123-456-7890'],
+            'city_categorical': ['New York', 'Madrid', 'New York'],
+            'example_default': [1, 2, 3],
+            'example_pii_true': [4, 5, 6],
+            'example_pii_false': [7, 8, 9],
+            'unknown_pii_true': ['a', 'b', 'c'],
+            'unknown_pii_false': ['a', 'b', 'c'],
+            'id_pii_true': ['ID_001', 'ID_002', 'ID_003'],
+            'id_pii_false': ['ID_001', 'ID_002', 'ID_003'],
+        })
+        metadata = SingleTableMetadata().load_from_dict({
+            'columns': {
+                'name_pii': {'sdtype': 'name'},
+                'phone_pii': {'sdtype': 'phone_number', 'pii': True},
+                'city_categorical': {'sdtype': 'city', 'pii': False},
+                'example_default': {'sdtype': 'example'},
+                'example_pii_true': {'sdtype': 'example', 'pii': True},
+                'example_pii_false': {'sdtype': 'example', 'pii': False},
+                'unknown_pii_true': {'sdtype': 'unknown', 'pii': True},
+                'unknown_pii_false': {'sdtype': 'unknown', 'pii': False},
+                'id_pii_true': {'sdtype': 'id', 'pii': True},
+                'id_pii_false': {'sdtype': 'id', 'pii': False},
+            },
+        })
+        dp = DataProcessor(metadata)
+        dp._transformers_by_sdtype['example'] = FloatFormatter()
+
+        # Run
+        config = dp._create_config(data, set())
+
+        # Assert
+        assert config['sdtypes'] == {
+            'example_default': 'example',
+            'unknown_pii_true': 'pii',
+            'phone_pii': 'pii',
+            'name_pii': 'pii',
+            'id_pii_true': 'pii',
+            'example_pii_false': 'example',
+            'unknown_pii_false': 'pii',
+            'id_pii_false': 'pii',
+            'example_pii_true': 'example',
+            'city_categorical': 'categorical'
+        }
+        expected_transformers = {
+            'example_default': FloatFormatter,
+            'unknown_pii_true': AnonymizedFaker,
+            'phone_pii': AnonymizedFaker,
+            'name_pii': AnonymizedFaker,
+            'id_pii_true': AnonymizedFaker,
+            'example_pii_false': FloatFormatter,
+            'unknown_pii_false': AnonymizedFaker,
+            'id_pii_false': AnonymizedFaker,
+            'example_pii_true': FloatFormatter,
+            'city_categorical': UniformEncoder
```

Comment:
for all the anonymized fakers, we should also test that the function_name is correctly assigned

Reply:
Yes, done in [d27e8d4](https://github.com/sdv-dev/SDV/pull/1782/commits/d27e8d45aa756c20d137a9b6fe7ce4990ba793f5)